frontend/src/pages/Compare.tsx

```diff
@@ -47,23 +58,68 @@ export default function Compare(props: CompareProps) {
+      onSuccess: data => {
+        // Set the version based on the runs included.
+        let version: CompareVersion = CompareVersion.InvalidRunCount;
+        if (data && data.length >= 2 && data.length <= 10) {
+          for (const run of data) {
+            const runVersion = run.run?.pipeline_spec?.hasOwnProperty('pipeline_manifest')
+              ? CompareVersion.V2
+              : CompareVersion.V1;
+            if (version === CompareVersion.InvalidRunCount) {
+              version = runVersion;
+            } else if (version !== runVersion) {
+              version = CompareVersion.Mixed;
+            }
+          }
+        }
+
+        // Update banner based on feature flag, run versions, and run count.
+        if (isFeatureEnabled(FeatureKey.V2_ALPHA) && version === CompareVersion.InvalidRunCount) {
+          props.updateBanner({
+            additionalInfo:
+              'At least two runs and at most ten runs must be selected to view the Run Comparison page.',
+            message:
+              'Error: failed loading the Run Comparison page. Click Details for more information.',
+            mode: 'error',
+          });
+        } else if (isFeatureEnabled(FeatureKey.V2_ALPHA) && version === CompareVersion.Mixed) {
+          props.updateBanner({
+            additionalInfo:
+              'The selected runs are a mix of V1 and V2.' +
+              ' Please select all V1 or all V2 runs to view the associated Run Comparison page.',
+            message:
+              'Error: failed loading the Run Comparison page. Click Details for more information.',
+            mode: 'error',
+          });
+        } else if (!isFeatureEnabled(FeatureKey.V2_ALPHA) && version === CompareVersion.V2) {
+          props.updateBanner({
+            additionalInfo:
+              'The selected runs are all V2, but the V2_ALPHA feature flag is disabled.' +
+              ' The V1 page will not show any useful information for these runs.',
+            message:
+              'Info: enable the V2_ALPHA feature flag in order to view the updated Run Comparison page.',
+            mode: 'info',
+          });
+        } else {
+          props.updateBanner({});
+        }
+
+        setCompareVersion(version);
+      },
     },
   );
 
-  if (isError || data === undefined) {
+  if (isError) {
     return <></>;
   }
 
-  // Display the V2 Compare page if all selected runs are V2.
-  if (
-    isFeatureEnabled(FeatureKey.V2_ALPHA) &&
-    isSuccess &&
-    data &&
-    data.every(run => run.run?.pipeline_spec?.hasOwnProperty('pipeline_manifest'))
-  ) {
+  if (isFeatureEnabled(FeatureKey.V2_ALPHA) && compareVersion === CompareVersion.V2) {
     return <CompareV2 />;
-  } else {
+  } else if (!isFeatureEnabled(FeatureKey.V2_ALPHA) || compareVersion === CompareVersion.V1) {
     return <CompareV1 {...props} />;
   }
+
+  // Only the error banner shown for mixed run versions or less than two selected runs.
```

Comment:
I understand that logic - if the banner might be wiped out anyway due to the structure of the `Compare` page, let's not show the `CompareV1` page at all.

I will note that the current banner for the case described is actually an _info_ banner that looks as follows:
![KFPV2InfoBanner](https://user-images.githubusercontent.com/7987279/172201068-53223dbc-c315-4202-a3be-c5cb448db8a4.png)

So the banner's current contents are a bit dependent on the v1 page being shown.

I'm not sure what the first line of your suggestion meant to indicate (the last bit of `InvalidRunCount.V1` isn't a variable but references both `CompareVersion.InvalidRunCount` and `CompareVersion.V1`); regardless, maybe the best option is  to show the V1 page with no banner in that case? I'm also fine showing an error and preventing them from doing this, my only hesitancy is you mentioned that disabling the feature flag is a deliberate action and they may have some reason for wanting to view the V1 page, even if it doesn't normally present any useful comparison.

Reply:
1. Ok got it, I'll look into this - I might remove the info banner altogether to avoid making overly complex changes, but I'll look into how to modify `CompareV1` first.
2. That makes perfect sense, I'll do that - thanks!
3. Got it, thanks for the update 👍